### PR TITLE
Docs/add accessibility intro

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes

This PR adds a general accessibility introduction page to to the Amplify UI Getting Started docs.  The intent is to showcase some of the accessibility related features of the design system and components.

#### Description of how you validated changes

Tested in local instance of Docs site.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/getting-started/accessibility/accessibility.angular.mdx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { Fragment } from '@/components/Fragment';
+
+<Fragment>
+  {({ platform }) => import(`./intro.mdx`)}
```
Comment (hbuchel) [inline docs/src/pages/[platform]/getting-started/accessibility/accessibility.angular.mdx:4]: Thoughts on adding more content for Vue/Angular? We don't have the primitives content to link to (and Authenticator doesn't currently have any accessibility specific content to link to I don't think)

File: docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+<Example>
+  <LabelInputExample />
+  <ExampleCode>
+```jsx file=./examples/LabelInputExample.tsx
+
+```
+  </ExampleCode>
+</Example>
+
+One use case for passing your own custom id to the field is if you need to target it for testing.
```
Comment (hbuchel) [inline docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx:32]: I'd like to add a small guide for automated testing with axe-core, maybe that can be a follow up.

File: docs/src/pages/[platform]/getting-started/accessibility/examples/AriaLiveExample.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  function handleClick() {
+    setCount(count + 1);
+  }
+
+  return (
+    <Flex>
+      <Button onClick={handleClick}>Add card to list.</Button>
+      <Badge aria-live="polite" alignSelf="center">
+        {`${count} cards in list`}
```
Comment (hbuchel) [inline docs/src/pages/[platform]/getting-started/accessibility/examples/AriaLiveExample.tsx:15]: Voiceover only announces the {count} and not the whole string unless we use a string literal here  which is neat and fun. 😒 Something about it being separated into different text leafs in the accessibility tree if we just use `{count} cards in list`.

File: docs/src/pages/[platform]/index.page.tsx [hunk 4]
```diff
@@ -293,7 +293,7 @@ const HomePage = ({ colorMode }) => {
                   WCAG
                 </Link>{' '}
                 and{' '}
-                <Link isExternal href="https://www.w3.org/TR/wai-aria-1.2/">
+                <Link isExternal href="https://www.w3.org/WAI/ARIA/apg/">
```
Comment (hbuchel) [inline docs/src/pages/[platform]/index.page.tsx:296]: I know we're updating the home page soon, but updating this link to the new one in the meantime.

File: docs/src/pages/[platform]/getting-started/accessibility/intro.mdx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Alert } from '@aws-amplify/ui-react';
+
+Amplify UI components follow [WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/) and
```
Comment (joebuono) [inline docs/src/pages/[platform]/getting-started/accessibility/intro.mdx:3]: Can we make these links open up in a new tab? Perhaps we can use our Link primitive and the `isExternal` prop

File: docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+## What's included?
+
+Amplify UI components have some accessibility needs baked in for ease of use. This includes auto-associating labels with form inputs, a default theme with accessible color contrast, and the flexibility to 
+enhance components to create your own accessible patterns.
+
+##  Accessible form fields
+
+### Form Labels
+
+All of Amplify UI's form field components allow passing a label to the input. If an ID is not supplied for the component, one is autogenerated for you so that the label can be programatically associated with the input.
```
Comment (joebuono) [inline docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx:21]: Do we allow or _require_ passing a label to the input?

File: docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  img={<MdOutlineCheckBox />}
+  href="/react/components/checkboxfield#css-styling"
+  title="Styling Checkbox inputs"
+  desc="Various ways to style the CheckboxField" />
+</Grid>
+
+## Themable design tokens
+
+Amplify UI uses design tokens for the color palette, which makes modifying the theme to meet your own color contrast goals more manageable. Amplify UI also provides tokens for properties such as border width and font size if you need to make your own customizations for higher contrast requirements.
+
```
Comment (joebuono) [inline docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx:63]: Perhaps we can include a small demo of using design tokens here

File: docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+ <CardLink 
+  img={<MdOutlineAutoAwesome />}
+  href="/react/components/textfield#accessibility--label-behavior"
+  title="Theming overview"
+  desc="Learn how to make a custom theme" />
+</Grid>
+
+## Enhancable primitives
+
+Since Amplify UI supports passing down HTML and ARIA attributes to primitive components, it's possible to create your own accessible patterns. The following example shows how to use a View component as an HTML nav element and how to add a unique name to the nav element using an `aria-label` attribute.
```
Comment (joebuono) [inline docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx:74]: nit: `<nav>`

File: docs/src/pages/[platform]/getting-started/accessibility/examples/AriaLiveExample.tsx (hunk 3 above)
Comment (joebuono) [reply docs/src/pages/[platform]/getting-started/accessibility/examples/AriaLiveExample.tsx:15]: Cool example!

**Review by joebuono — APPROVED** [review]
Looks great, just a few small comments!

File: docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx (hunk 6 above)
Comment (hbuchel) [reply docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx]: Good point, updated!

File: docs/src/pages/[platform]/getting-started/accessibility/intro.mdx (hunk 5 above)
Comment (hbuchel) [reply docs/src/pages/[platform]/getting-started/accessibility/intro.mdx]: Yeah, that makes sense. Updated in latest.

File: docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx (hunk 8 above)
Comment (hbuchel) [reply docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx]: Thanks, fixed in latest.

File: docs/src/pages/[platform]/getting-started/accessibility/index.page.mdx [hunk 9]
```diff
@@ -0,0 +1,10 @@
+---
+title: Accessibility
+metaTitle: Accessibility
+description: An introduction to building accessible apps with Amplify UI.
```
Comment (zchenwei) [inline docs/src/pages/[platform]/getting-started/accessibility/index.page.mdx:4]: Do we need to add a `metaDescription`? I saw that exist in other pages but not sure why we duplicate it with `description`

File: docs/src/pages/[platform]/getting-started/accessibility/examples/AriaLiveExample.tsx (hunk 3 above)
Comment (zchenwei) [reply docs/src/pages/[platform]/getting-started/accessibility/examples/AriaLiveExample.tsx:15]: Wow, good to know. Using `aria-atomic` is wired in this case

File: docs/src/pages/[platform]/getting-started/accessibility/accessibility.angular.mdx (hunk 1 above)
Comment (zchenwei) [reply docs/src/pages/[platform]/getting-started/accessibility/accessibility.angular.mdx:4]: Yea, I think it should be good for now

File: docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+import { MdOutlineCheckBox, MdEdit, MdOutlineAutoAwesome } from 'react-icons/md';
+
+<Fragment>
+  {({ platform }) => import(`./intro.mdx`)}
+</Fragment>
+
+## What's included?
+
+Amplify UI components have some accessibility needs baked in for ease of use. This includes auto-associating labels with form inputs, a default theme with accessible color contrast, and the flexibility to 
+enhance components to create your own accessible patterns.
```
Comment (zchenwei) [inline docs/src/pages/[platform]/getting-started/accessibility/accessibility.react.mdx:15]: Is this forced to break onto 2 lines?

File: docs/src/pages/[platform]/getting-started/accessibility/index.page.mdx (hunk 9 above)
Comment (hbuchel) [reply docs/src/pages/[platform]/getting-started/accessibility/index.page.mdx:4]: description will also be used as the page meta description in the `<head>`, but it also outputs it in the UI (beneath the title in a larger font).

So, use metaDescription if you don't want it visible on the page, or use description if you want the same text in both places and visible on the page, or use both if for some reason you want the visible text to be different than the meta description.

**Review by zchenwei — APPROVED** [review]
This is looking awesome! Thanks for adding it :)

File: docs/src/pages/[platform]/getting-started/accessibility/index.page.mdx (hunk 9 above)
Comment (zchenwei) [reply docs/src/pages/[platform]/getting-started/accessibility/index.page.mdx:4]: Got it, I am left behind in docs :(

**Pull request merged**